### PR TITLE
Fix #59364 : Respect global .gitignore when searching

### LINKED ISSUE #59364: Respect global .gitignore when searching
The global .gitignore file, aka git config `core.excludesfile` setting should be used when searching, just like the local .gitignore file is.

Currently files/folders will be ignored if they are in the local .gitignore, but not if they are only in the global git ignore.

#49239 asked about this feature, but it was incorrectly closed.

### PR DESCRIPTION
**Pull request by skprabhanjan**
@roblourens , Here is the PR that fixes #59364 .
Please review the same and let me know if any changes are needed.
Thanks :)

### CODE REVIEW

**Comment by roblourens** [comment]
This is a good start, but it needs to be hooked up to a setting so users can use it :)

Follow the example of the `search.useIgnoreFiles` setting, and follow the same naming convention.

Then the flag on the query also needs to be checked in the extension version of search - https://github.com/Microsoft/vscode/blob/master/extensions/search-rg/src/ripgrepTextSearch.ts#L350

**Comment by skprabhanjan** [comment]
@roblourens , Thanks for the review :) 
I will work on these and get it fixed.

**Comment by skprabhanjan** [comment]
@roblourens , Please check the changes now.
I have added a setting and also used the same naming convention.
My doubt was how to use that setting in the https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/search/node/ripgrepTextSearch.ts#L514 , where I had used a config property in IRawSearch interface(search.useIgnoreFiles example was not used here), I have kept that same as before, please let me know if that is fine or I need to change.
Thanks :)

**Review by roblourens — CHANGES_REQUESTED** [review]
You do need to use the setting somewhere, currently it is not hooked up to anything. For example the useIgnoreFiles setting is used here: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/search/common/queryBuilder.ts#L63

**Comment by skprabhanjan** [comment]
@roblourens , I have followed the exact way that search.useIgnoreFiles was implemented.(Taking into account both the review feedbacks that you gave).
Please review this and let me if there are any changes :)
Thanks for the constant feedback and help :)

**Comment by skprabhanjan** [comment]
> It looks like the build has an issue

I checked that and all the errors are related to test files and few of them are due to the "disregardGlobalIgnoreFiles" flag that was added and others were related to something else, I did not know if I had to fix that build issue.
So what should I do in this regard?
Thanks :)

**Comment by skprabhanjan** [comment]
Hey @roblourens , sorry for that build failure error, it was my bad.
I fixed it now, can you please check it once again :)
Thanks :)

File: src/vs/workbench/services/search/node/ripgrepTextSearch.ts [hunk 1]
```diff
@@ -511,7 +511,9 @@ function getRgArgs(config: IRawSearch) {
 	}
 
 	args.push('--no-config');
-	args.push('--no-ignore-global');
+	if (!config.disregardGlobalIgnoreFiles) {
+		args.push('--no-ignore-global');
```
Comment (roblourens) [inline src/vs/workbench/services/search/node/ripgrepTextSearch.ts]: I think this is switched around? I know the double negative in our naming convention makes it a bit hard to reason about but if we want to disregard the global ignore files, then we should add the "no" flag... so I think the `!` needs to be removed.

**Comment by roblourens** [comment]
I added one comment to the code but everything else looks good. But the QueryBuilder tests are still failing because now this extra property is added to the query. So could you just update them so they pass?

I'm not sure about the CompletionModel tests. You could merge the latest from master into your PR branch to check whether it's something else from master or your code.

**Comment by skprabhanjan** [comment]
@roblourens , Fixed all the changes and QueryBuilder tests are now working.
Also I ran the ./scripts/test.sh locally and did not find anything failing.
Thanks :)

**Review by roblourens — APPROVED** [review]
Great work, thanks!

**Pull request merged**